addons/mail/static/src/js/mail.js

```diff
@@ -35,6 +35,20 @@ openerp.mail = function (session) {
             return [text, false];
         },
 
+        /* If there is unread message in thread, displays Mark all read button, else hide it.
+        */
+        show_mark_all_read : false,
+
+        check_message_read: function(){
```

Comment:
Same thing here. Chatter utils holds some tools methods for the chatter. This method should be a class method for the wall / thread manager.